eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/immutable/AbstractImmutableBagIterable.java

```diff
@@ -67,7 +67,8 @@
             Function2<? super V, ? super T, ? extends V> nonMutatingAggregator)
     {
         MutableMap<K, V> map = UnifiedMap.newMap();
-        this.forEach(new NonMutatingAggregationProcedure<>(map, groupBy, zeroValueFactory, nonMutatingAggregator));
+        NonMutatingAggregationProcedure<T, K, V> nonMutatingAggregationProcedure = new NonMutatingAggregationProcedure<>(map, groupBy, zeroValueFactory, nonMutatingAggregator);
```

Comment:
I would shorten the variable name to procedure.

Reply:
Done.